app/keepers/keepers.go

```diff
@@ -421,6 +445,10 @@ func NewAppKeepers(
 	appKeepers.ScopedICAControllerKeeper = scopedICAControllerKeeper
 	appKeepers.ScopedInterTxKeeper = scopedInterTxKeeper
 
+	// set the contract keeper for the Ics20WasmHooks
+	appKeepers.ContractKeeper = wasmkeeper.NewDefaultPermissionKeeper(appKeepers.WasmKeeper)
```

Comment:
So these hooks are governed by permissioned contracts? Trying to understand how this works in a permissionless smart contract context.

Reply:
`NewDefaultPermissionKeeper` returns a wrapper around the wasm keeper with the "default" permissions. But the permissions here only matter for create, instantiate(2), migrate, UpdateContractAdmin, ClearContractAdmin, and SetAccessConfig. None of which we're using.

Even if we were using those, I think the default permissions is what we want. It's used in places like [this](https://github.com/CosmWasm/wasmd/blob/558147d054c88840a804a3ec4e0bba5b3cb5905a/x/wasm/keeper/keeper.go#L190-L203) and define like [this](https://github.com/CosmWasm/wasmd/blob/558147d054c88840a804a3ec4e0bba5b3cb5905a/x/wasm/keeper/authz_policy.go#L27-L45). Plus, the only other policy is "gov" which just allows everything (and we still want, for example, that only admins can migrate contracts).

An alternative here would be to write our own wrapper that implements [ContractOpsKeeper](https://github.com/CosmWasm/wasmd/blob/558147d054c88840a804a3ec4e0bba5b3cb5905a/x/wasm/types/exported_keepers.go#L28) that panics on the functions we're not using, but I don't see the point when wasmd already implements this in [PermissionedKeeper](https://github.com/CosmWasm/wasmd/blob/558147d054c88840a804a3ec4e0bba5b3cb5905a/x/wasm/keeper/contract_keeper.go#L37).